### ISSUE TITLE
[Bug]: Error deleting record: database disk image is malformed

### ISSUE THREAD

**Issue by keli5** [sqlitebrowser/sqlitebrowser#3433]
### What did you do?

I have a database that has a few known bad rows (some String columns have invalid, BLOB data), I think due to a failing disk. I've moved the database files onto a known good drive, but I cannot modify or delete these bad rows. The rest of the database can be edited and seems to be perfectly readable, I just need to fix this data on a couple of rows.

### What did you expect to see?

I expected that I would be able to modify or delete these malformed rows, or at least have an option to somehow force delete/modify with a "malformed" database.

### What did you see instead?

I simply get "Error changing data: database disk image is malformed" error, and cannot modify the rows I actually need to change.

### DB4S Version

3.12.2

### What OS are you seeing the problem on?

Windows

### OS version

Windows 10 22H2

### Relevant log output

```shell
SQL LOG:
(1) QBuffer::seek: Invalid pos: -4096 (, :0)
(1) QBuffer::seek: Invalid pos: -4096 (, :0)
(779) index corruption at line 91877 of [1b256d97b5]
(779) statement aborts at 28: [UPDATE "main"."GuildMember" SET "id"=? WHERE "_rowid_"='1'] database disk image is malformed
(1) updateRecord:  "database disk image is malformed" (, :0)
(779) index corruption at line 91877 of [1b256d97b5]
(779) statement aborts at 28: [UPDATE "main"."GuildMember" SET "id"=? WHERE "_rowid_"='1'] database disk image is malformed
(1) updateRecord:  "database disk image is malformed" (, :0)
(779) index corruption at line 91877 of [1b256d97b5]
(779) statement aborts at 28: [DELETE FROM "main"."GuildMember" WHERE _rowid_ IN ('1');] database disk image is malformed
(1) executeSQL:  "database disk image is malformed (DELETE FROM \"main\".\"GuildMember\" WHERE _rowid_ IN ('1');)" (, :0)
(1) deleteRecord:  "database disk image is malformed (DELETE FROM \"main\".\"GuildMember\" WHERE _rowid_ IN ('1');)" (, :0)
(779) index corruption at line 91877 of [1b256d97b5]
(779) statement aborts at 28: [DELETE FROM "main"."GuildMember" WHERE _rowid_ IN ('1');] database disk image is malformed
(1) executeSQL:  "database disk image is malformed (DELETE FROM \"main\".\"GuildMember\" WHERE _rowid_ IN ('1');)" (, :0)
(1) deleteRecord:  "database disk image is malformed (DELETE FROM \"main\".\"GuildMember\" WHERE _rowid_ IN ('1');)" (, :0)
(779) index corruption at line 91877 of [1b256d97b5]
(779) statement aborts at 28: [DELETE FROM "main"."GuildMember" WHERE _rowid_ IN ('1');] database disk image is malformed
(1) executeSQL:  "database disk image is malformed (DELETE FROM \"main\".\"GuildMember\" WHERE _rowid_ IN ('1');)" (, :0)
(1) deleteRecord:  "database disk image is malformed (DELETE FROM \"main\".\"GuildMember\" WHERE _rowid_ IN ('1');)" (, :0)
(779) index corruption at line 91877 of [1b256d97b5]
(779) statement aborts at 28: [UPDATE "main"."GuildMember" SET "id"=? WHERE "_rowid_"='1'] database disk image is malformed
(1) updateRecord:  "database disk image is malformed" (, :0)
```

### Prevention against duplicate issues

- [X] I have searched for similar issues

**Comment by justinclift** [comment]
Hmmm, you're probably best asking for assistance on the SQLite Forum:

https://sqlite.org/forum/forum

There are all kinds of specialist SQLite recovery tools around, so pinging the people there will probably get you the best possible result. :smile:

**Comment by keli5** [comment]
I'll look around there for a specialized tool then, thank you!
